### PR TITLE
VC: various fixes

### PR DESCRIPTION
**Pull request by cheatfate**
1. Fix timeouts for firstSuccess() template.
2. Fix logging statements to be compatible with beacon_node.
3. Add unexpected error handlers.
4. Fix block proposal issue for blocks from the past and for blocks from the future.

### CODE REVIEW

File: beacon_chain/validator_client/api.nim [hunk 1]
```diff
@@ -342,52 +342,57 @@ template firstSuccessTimeout*(vc: ValidatorClientRef, respType: typedesc,
                        RestBeaconNodeStatus.Uninitalized}
     let offlineNodes = vc.beaconNodes.filterIt(it.status in offlineMask)
 
-    warn "There no beacon nodes available, refreshing nodes status",
+    warn "There no working beacon nodes available, refreshing nodes status",
```
Comment (arnetheduck) [inline beacon_chain/validator_client/api.nim]:
```suggestion
    warn "No working beacon nodes available, refreshing node status",
```

File: beacon_chain/validator_client/block_service.nim [hunk 2]
```diff
@@ -55,28 +55,33 @@ proc publishBlock(vc: ValidatorClientRef, currentSlot, slot: Slot,
...
+          error "An unexpected error occurred while publishing block",
+                err_name = exc.name, err_msg = exc.msg
           return
       if res:
-        notice "Block published", slot = currentSlot,
-          validator = validator.pubKey, validator_index = validator.index.get(),
-          deposits = len(signedBlock.message.body.deposits),
-          attestations = len(signedBlock.message.body.attestations),
-          graffiti = graffiti
+        notice "Block proposed", blck = shortLog(signedBlock.message),
```
Comment (arnetheduck) [inline beacon_chain/validator_client/block_service.nim]:
```suggestion
        notice "Block published", blck = shortLog(signedBlock.message),
```

as long as they start with `Block `, should be ok - this way one can grep for just `Block ` and we can use "published" to find anything the VC does (then it should be "Attestation published" as well)

the `validator_index` is redundant, it should be logged in the block as `proposer_index`

File: beacon_chain/validator_client/fork_service.nim [hunk 3]
```diff
@@ -10,10 +10,14 @@ proc pollForFork(vc: ValidatorClientRef) {.async.} =
...
       return
+    except CatchableError as exc:
+      error "Unexpected error occured while getting fork information",
+            err_name = exc.name, err_msg = exc.msg
+      return
 
   if vc.fork.isNone() or vc.fork.get() != fork:
     vc.fork = some(fork)
-    notice "Fork update success", fork = fork
+    debug "Fork update success", fork = fork
```
Comment (arnetheduck) [inline beacon_chain/validator_client/fork_service.nim]: This can remain `notice` - it should happen .. almost never and is quite important

**Review by arnetheduck — APPROVED** [review]

File: beacon_chain/validator_client/api.nim (hunk 1 above)
Comment (cheatfate) [reply beacon_chain/validator_client/api.nim]: Fixed.

File: beacon_chain/validator_client/block_service.nim (hunk 2 above)
Comment (cheatfate) [reply beacon_chain/validator_client/block_service.nim]: Fixed.

File: beacon_chain/validator_client/fork_service.nim (hunk 3 above)
Comment (cheatfate) [reply beacon_chain/validator_client/fork_service.nim]: Fixed.

**Pull request merged**